Robust_Lab_Vartiainen.qmd

```diff
@@ -0,0 +1,348 @@
+    my_penguins
+    ```
+
+a\. Visualize body size by sex
+
+```{r}
+ggplot(my_penguins, aes(sex, body_mass_g, fill=sex)) +
+geom_boxplot()+geom_point()
+```
+
+b\. Calculate the original mean difference between sex\
+
+```{r}
+mean_diff <- my_penguins %>% 
+  group_by(sex) %>%
+  summarise(mean_group = mean(body_mass_g)) %>%
+  summarise(mean_diff=diff(mean_group))
+
+```
+
+c\. Permute the group labels (10000x)
+
+```{r}
+sample_size <- nrow(my_penguins)
+perm_reps   <- 10000 
+many.perm <- my_penguins    %>%
+  rep_sample_n(size = sample_size, replace = FALSE, reps = perm_reps) %>% 
+  mutate(perm_sex = sample(sex, size = n(), replace = FALSE))  %>%
+  group_by(replicate, perm_sex)
+many.perm
+```
+
+d\. Plot the null-hypothesis distribution (NHD) for the difference
+
+```{r}
+many.perm.means <- many.perm %>%
+  summarise(mean_group = mean(body_mass_g), .groups = "drop")%>%
+  group_by(replicate)
+many.perm.means
+many.perm.diffs <- many.perm.means %>%
+  summarise(diff_value = diff(mean_group))
+ggplot(many.perm.diffs, aes(x = diff_value)) +
+  geom_histogram(bins = 32, color = "white")
+
+
+df_diff  <- my_penguins          %>% 
+  specify(body_mass_g ~ sex) %>%
+  calculate(stat = "diff in means")
+  null_distn  <- my_penguins   %>% 
+  specify(body_mass_g ~ sex) %>%
+   hypothesize(null = "independence") %>%
+   generate(reps = 10000, type = "permute") %>%
+   calculate(stat = "diff in means")
+  null_distn %>%
+  visualize() +shade_p_value(obs_stat = df_diff, direction = "two-sided")
+```
+
+e\. Compare the observed mean difference to the NHD (is *p* \< .05?)
+
+Yes!
+
+```{r}
+many.perm.diffs_1 <- many.perm.diffs %>% 
+  mutate(abs_obs_dif = abs(pull(mean_diff)),
+         abs_perm_dif = abs(diff_value),
+         as_or_more_extreme = abs_perm_dif >= abs_obs_dif) 
+mean(many.perm.diffs_1$as_or_more_extreme)
+
+
+```
+
+6.  Suppose a replication experiment was conducted to further examine the interaction effect between driving difficulty and conversation difficulty on driving errors in a driving simulator. In the replication, the researchers administered the same three levels of conversation difficulty; (1) control, (2) easy, (3) difficult (C, E, D) but assume that they added a third level of driving difficulty; (1) low, (2) moderate, (3) difficult (L, M, D). Assume the design was completely between subjects and conduct a factorial ANOVA to test the main effects of conversation and driving difficulty as well as the interaction effect. The DV is the number of errors committed in the driving simulator.
+
+    ```{r}
+    library(tidyverse)
+    fac_data<-read_csv("https://raw.githubusercontent.com/jgeller112/psy503-psych_stats/master/static/assignment/data/fact_final.csv")
+
+    ```
+
+    a\. Run a permutation test (ANOVA)
+
+    ```{r}
+    anova_perm <- aovperm(errors ~ convo + drive, data = fac_data, np = 10000)
+    anova_perm
+
+    ```
+
+    b\. How would you follow-up significant effects in this context?
+
```

Comment:
I would run bootstraps on the difference score between comparisons of interest. This will give us 95% cis which we can use to make inferences. 

An argument can be made for permutation tests, but this will just tell us that a difference exists.